hveto/cli/weekly_hveto.py

```diff
@@ -0,0 +1,407 @@
+    """)
+
+    epilog = textwrap.dedent("""
+    Running with default (no) arguments  will run hveto over 7 days of data ending at 00:00 UTC this morning.
+
+    To run hveto for every day for the month of July, each run covering 7 days:
+
+        python -m weekly_hveto --start 7/1 --end 7/31 --duration 7 --stride 1
+
+    """)
+    parser = argparse.ArgumentParser(description=description, epilog=epilog, prog=__process_name__,
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser_add_args(parser)
+    args = parser.parse_args()
+    verbosity = 0 if args.quiet else args.verbose
+
+    if verbosity < 1:
+        logger.setLevel(logging.CRITICAL)
+    elif verbosity < 2:
+        logger.setLevel(logging.INFO)
+    else:
+        logger.setLevel(logging.DEBUG)
+
+    output_directory = Path(args.output_directory)
+    now_time = datetime.now(timezone.utc).strftime('%Y%m%d-%H%M%S')
+    dag_dir = output_directory / f'DAG-{now_time}'
+    dag_dir.mkdir(parents=True, exist_ok=True)
+
+    dag_log = dag_dir / 'weekly-dag.log'
+    dag_file = dag_dir / 'hveto.dag'
+
+    # add logging too a file
+    NOW = datetime.now()
+    TIMEZONE = reference.LocalTimezone().tzname(NOW)
+    DATEFMT = '%Y-%m-%d %H:%M:%S {}'.format(TIMEZONE)
+    FMT = '%(name)s %(asctime)s %(levelname)+8s: %(filename)s:%(lineno)d:  %(message)s'
+    logf_formatter = logging.Formatter(FMT, datefmt=DATEFMT)
+    logf_handler = logging.FileHandler(dag_log, mode='w')
+    logf_handler.setFormatter(logf_formatter)
+    logger.addHandler(logf_handler)
+
+    # debugging?
+    logger.debug(f'{__process_name__} version: {__version__} called with arguments:')
+    for k, v in args.__dict__.items():
+        logger.debug('    {} = {}'.format(k, v))
+
+    config = args.config_file
+    if config is None:
+        logger.critical('Configuration file not specified')
+        raise ValueError('Configuration file not specified')
+
+    if args.ifo == 'UK':
+        logger.critical('IFO must be specified on command line or in the environment, if not running at LLO or LHO')
+        raise ValueError('IFO must be specified')
+    elif args.ifo not in ['L1', 'H1']:
+        logger.critical(f'IFO {args.ifo} not in ["L1", "H1"]')
+        raise ValueError(f'IFO {args.ifo} not in ["L1", "H1"]')
+
+    start_day = args.start
+    start_gps = to_gps(start_day)
+    start_dt = from_gps(start_gps)
+
+    end_day = args.end
+    end_gps = to_gps(end_day)
+    end_dt = from_gps(end_gps)
+    next_dt = end_dt
+
+    duration = args.duration
+    duration_dt = timedelta(days=duration)
+
+    stride = args.stride
+    stride_dt = timedelta(days=stride)
+
+    logger.info(f'DAG file  will be written to {dag_file}')
+    dag_file.parent.mkdir(parents=True, exist_ok=True)
+
+    with dag_file.open('w') as dag_fh:
+        print(f'# process long duration hveto for {start_day} to {end_day} each {duration} days', file=dag_fh)
```

Comment:
@areeda I think you might have already fixed this during our meeting, but I think this should be something like '# process long duration hveto for {start_day} to {end_day}, running each {stride} days, with each run covering {duration} days' - but note that it could confuse the user a little, in that there is no run that includes 'end_day' (here and throughout the code).